src/catalog/sequence_catalog.cpp

```diff
@@ -0,0 +1,300 @@
+  if (result_tiles == nullptr || result_tiles->size() == 0) {
+    LOG_INFO("no sequence on database %d and %s", database_oid,
+             sequence_name.c_str());
+    return std::shared_ptr<SequenceCatalogObject>(nullptr);
+  } else {
+    LOG_INFO("size of the result tiles = %lu", result_tiles->size());
+  }
+
+  PELOTON_ASSERT(result_tiles->size() == 1);
+  size_t tuple_count = (*result_tiles)[0]->GetTupleCount();
+  PELOTON_ASSERT(tuple_count == 1);
+  auto new_sequence = std::make_shared<SequenceCatalogObject>(
+      (*result_tiles)[0]->GetValue(0, 0).GetAs<oid_t>(),
+      (*result_tiles)[0]->GetValue(0, 1).ToString(),
+      (*result_tiles)[0]->GetValue(0, 2).GetAs<int64_t>(),
+      (*result_tiles)[0]->GetValue(0, 3).GetAs<int64_t>(),
+      (*result_tiles)[0]->GetValue(0, 4).GetAs<int64_t>(),
+      (*result_tiles)[0]->GetValue(0, 5).GetAs<int64_t>(),
+      (*result_tiles)[0]->GetValue(0, 6).GetAs<bool>(),
+      (*result_tiles)[0]->GetValue(0, 7).GetAs<int64_t>(), txn);
+
+  return new_sequence;
+}
+
+bool SequenceCatalog::UpdateNextVal(oid_t sequence_oid, int64_t nextval,
+    concurrency::TransactionContext *txn){
+  std::vector<oid_t> update_columns({SequenceCatalog::ColumnId::SEQUENCE_VALUE});
+  std::vector<type::Value> update_values;
+  update_values.push_back(type::ValueFactory::GetBigIntValue(nextval).Copy());
+  std::vector<type::Value> scan_values;
+  scan_values.push_back(type::ValueFactory::GetIntegerValue(sequence_oid).Copy());
+  oid_t index_offset = SequenceCatalog::IndexId::PRIMARY_KEY;
+
+  return UpdateWithIndexScan(update_columns, update_values, scan_values, index_offset, txn);
+}
+
+/* @brief   get sequence oid from pg_sequence table given sequence_name and
+ * database_oid
+ * @param   database_oid  the databse_oid associated with the sequence
+ * @param   sequence_name the name of the sequence
+ * @param   txn           current transaction
+ * @return  the oid_t of the sequence if the sequence is found, INVALID_OID
+ * otherwise
+ */
+oid_t SequenceCatalog::GetSequenceOid(std::string sequence_name,
+                                      oid_t database_oid,
+                                      concurrency::TransactionContext *txn) {
+  std::vector<oid_t> column_ids({ColumnId::SEQUENCE_OID});
+  oid_t index_offset = IndexId::DBOID_SEQNAME_KEY;
+  std::vector<type::Value> values;
+  values.push_back(type::ValueFactory::GetIntegerValue(database_oid).Copy());
+  values.push_back(type::ValueFactory::GetVarcharValue(sequence_name).Copy());
+
+  // the result is a vector of executor::LogicalTile
+  auto result_tiles =
+      GetResultWithIndexScan(column_ids, index_offset, values, txn);
+  // carefull! the result tile could be null!
+  if (result_tiles == nullptr || result_tiles->size() == 0) {
+    LOG_INFO("no sequence on database %d and %s", database_oid,
+             sequence_name.c_str());
+    return INVALID_OID;
+  }
+
+  PELOTON_ASSERT(result_tiles->size() == 1);
+  oid_t result;
```

Comment:
Since the variable name "result" does not give any further information, how about compressing lines 293-296 into one line:
return (*result_tiles)[0]->GetValue(0, 0).GetAs<oid_t>();